### PR TITLE
Add support for rook 1.9

### PR DESCRIPTION
**Pull request by emosbaugh**
Updates Rook go dependency to 1.9.12.
Increases Rook manager count to 2 as the cluster scales.
Sets Ceph CSI provisioner and plugin resources once cluster is scaled to 3 nodes as it should have sufficient capacity.

### CODE REVIEW

File: pkg/cluster/rook_ceph.go [hunk 1]
```diff
@@ -233,14 +233,115 @@ func (c *Controller) ReconcileMonCount(count int) error {
...
+
+	if cluster.Spec.Mgr.Count == count {
+		return nil
+	}
+	if cluster.Spec.Mgr.Count > count {
+		c.Log.Debugf("Will not reduce mgr count from %s to %s", cluster.Spec.Mgr.Count, count)
+		return nil
+	}
+
+	c.Log.Infof("Changing mgr count from %d to %d", cluster.Spec.Mgr.Count, count)
```
Comment (laverya) [inline pkg/cluster/rook_ceph.go:269]:
```suggestion
	c.Log.Infof("Increasing mgr count from %d to %d", cluster.Spec.Mgr.Count, count)
```

File: pkg/ekcoops/operator.go [hunk 2]
```diff
@@ -127,19 +127,32 @@ func (o *Operator) reconcileRook(nodes []corev1.Node, doFullReconcile bool) erro
 		readyCount, err := o.ensureAllUsedForStorage(nodes)
 		if err != nil {
 			multiErr = multierror.Append(multiErr, errors.Wrapf(err, "ensure all ready nodes used for storage"))
-		} else {
+		}
+
+		if readyCount > 0 {
```
Comment (laverya) [inline pkg/ekcoops/operator.go:132]: Why did you change this? It seems to me like you're relying on the return value of a function that returned error now?

File: pkg/cluster/rook_ceph.go [hunk 3]
```diff
@@ -233,14 +233,115 @@ func (c *Controller) ReconcileMonCount(count int) error {
 
 	c.Log.Infof("Changing mon count from %d to %d", cluster.Spec.Mon.Count, count)
 	cluster.Spec.Mon.Count = count
-	_, err = c.Config.CephV1.CephClusters("rook-ceph").Update(context.TODO(), cluster, metav1.UpdateOptions{})
```
Comment (rrpolanco) [inline pkg/cluster/rook_ceph.go:236]: Are we planning to use `context` in some meaningful way? If not, I'd rather use `context.TODO()` where required only.

Comment (emosbaugh) [reply pkg/cluster/rook_ceph.go:236]: I moved the context.TODO() call up the stack. I'm punting on making a decision what context to use in the call to this function. It is my intention to eventually use context from the process.

File: pkg/cluster/rook_ceph.go (hunk 1 above)
Comment (emosbaugh) [reply pkg/cluster/rook_ceph.go]: Changed

**Review by laverya — DISMISSED** [review]

File: pkg/ekcoops/operator.go [hunk 4]
```diff
@@ -127,19 +127,32 @@ func (o *Operator) reconcileRook(nodes []corev1.Node, doFullReconcile bool) erro
...
+				err := o.controller.ReconcileMonCount(context.TODO(), readyCount)
 				if err != nil {
 					multiErr = multierror.Append(multiErr, errors.Wrapf(err, "reconcile mon count"))
 				}
+				err = o.controller.ReconcileMgrCount(context.TODO(), readyCount)
+				if err != nil {
+					multiErr = multierror.Append(multiErr, errors.Wrapf(err, "reconcile mgr count"))
+				}
 			}
 		}
```
Comment (rrpolanco) [inline pkg/ekcoops/operator.go:146]: Why not just put this code in the first `if err !=nil` block?

Comment (emosbaugh) [reply pkg/ekcoops/operator.go]: Oops. It should have been `if err == nil {`. Fixed

**Review by laverya — APPROVED** [review]
